test/replication-luatest/bootstrap_strategy_test.lua

```diff
@@ -179,3 +179,210 @@ g_auto.test_sync_waits_for_all_connected = function(cg)
     end, {cg.box_cfg.replication})
     cg.replica:assert_follows_upstream(1)
 end
+
+local g_config = t.group('gh-7999-bootstrap-strategy-config')
+
+g_config.after_each(function(cg)
+    cg.replica_set:clean()
+end)
+
+g_config.before_test('test_no_replication', function(cg)
+    cg.replica_set = replica_set:new{}
+    cg.server1 = cg.replica_set:build_and_add_server{
+        alias = 'server1',
+        box_cfg = {
+            replication_timeout = 0.1,
+            bootstrap_strategy = 'config',
+            bootstrap_leader = server.build_listen_uri('server1'),
+            replication = nil
+        },
+    }
+end)
+
+local no_leader_msg = 'failed to connect to the bootstrap leader'
+
+g_config.test_no_replication = function(cg)
+    cg.replica_set:start{wait_until_ready = false}
+    local logfile = fio.pathjoin(cg.server1.workdir, 'server1.log')
+    t.helpers.retrying({}, function()
+        t.assert(server:grep_log(no_leader_msg, nil, {filename = logfile}))
+    end)
+end
+
+g_config.before_test('test_uuid', function(cg)
+    cg.replica_set = replica_set:new{}
+    cg.server1 = cg.replica_set:build_and_add_server{
+        alias = 'server1',
+        box_cfg = {
+            bootstrap_strategy = 'config',
+            bootstrap_leader = uuid1,
+            instance_uuid = uuid1,
+            replication = nil,
+        },
+    }
+end)
+
+g_config.after_test('test_uuid', function(cg)
```

Comment:
Can this be moved after the test itself? Right now it is kinda awkward to read `before_test`->`after_test`->`test`. Same for other tests as well, won't mention it there.

Reply:
Sure. Done, won't post the diff since it's trivial.